src/system/SystemObject.h

```diff
@@ -204,9 +197,78 @@ class ObjectPool
 public:
     void Reset();
 
-    T * TryCreate();
     void GetStatistics(chip::System::Stats::count_t & aNumInUse, chip::System::Stats::count_t & aHighWatermark);
 
+    template <typename... Args>
+    T * TryCreate(Args &&... args)
+    {
+        T * lReturn = nullptr;
+        static_assert(std::is_base_of<Object, T>::value, "Try to create an object which is not derived from class Object");
+
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+        T * newNode = new T(std::forward<Args>(args)...);
+
+        if (newNode->TryCreate())
```

Comment:
Just as as matter of codesize, this part of the method, which does not depend on `Args`, should probably be a separate function that is just templated on `T`, not `Args`?  Or even a non-template out-of-line function, since it really just depends on `Object`?

Doing this refactoring ins a followup is ok, just to keep the scope sane here.